Lib/ast.py

```diff
@@ -702,11 +703,20 @@ def maybe_newline(self):
         if self._source:
             self.write("\n")
 
-    def fill(self, text=""):
+    def maybe_semicolon(self):
+        """Adds a "; " delimiter if it isn't the start of generated source"""
+        if self._source:
+            self.write("; ")
+
+    def fill(self, text="", allowsemi=True):
         """Indent a piece of text and append it, according to the current
-        indentation level"""
-        self.maybe_newline()
-        self.write("    " * self._indent + text)
+        indentation level, or only delineate with semicolon if applicable"""
+        if self._in_interactive and not self._indent and allowsemi:
+            self.maybe_semicolon()
+            self.write(text)
```

Comment:
If the text is empty, do we have a trailing whitespace?

Reply:
The actual `text` param passed to `fill()` can be empty but as far as I see there will always be other text that follows as this is called before adding new things, never at the end of something.